x/compute/internal/keeper/keeper.go

```diff
@@ -892,16 +892,24 @@ func consumeGas(ctx sdk.Context, gas uint64) {
 }
 
 // generates a contract address from codeID + instanceID
-func (k Keeper) generateContractAddress(ctx sdk.Context, codeID uint64) sdk.AccAddress {
+func (k Keeper) generateContractAddress(ctx sdk.Context, codeID uint64, creator sdk.AccAddress) sdk.AccAddress {
 	instanceID := k.autoIncrementID(ctx, types.KeyLastInstanceID)
-	return contractAddress(codeID, instanceID)
+	return contractAddress(codeID, instanceID, creator)
 }
 
-func contractAddress(codeID, instanceID uint64) sdk.AccAddress {
+func contractAddress(codeID, instanceID uint64, creator sdk.AccAddress) sdk.AccAddress {
 	// NOTE: It is possible to get a duplicate address if either codeID or instanceID
```

Comment:
I know this is Assaf's but probably this comment is no longer relevant, since even though the codeId might overflow, the hash with the creator will still make the address different

Reply:
@assafmo

Reply:
fe161c855